Add compile and link options echo to configure

jonasschnelli opened the pull request:
Adds some compile and link options output to the configure process.
Maybe I'm old school, but I like this level of verbosity.

Results in something like
```
[ ---snip--- ]
configure: Using bignum implementation: no
configure: Using scalar implementation: 64bit
configure: Using endomorphism optimizations: no
configure: Building ECDH module: no
configure: Building Schnorr signatures module: no
configure: Building ECDSA pubkey recovery module: yes
configure: Using jni: no
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating libsecp256k1.pc
config.status: creating src/libsecp256k1-config.h
config.status: executing depfiles commands
config.status: executing libtool commands

Options used to compile and link:
  WITH WALLET   = yes
  WITH GUI / QT = yes
   WITH QR      = yes
  WITH TEST     = no
  WITH BENCH    = yes

  TARGET OS     = darwin
  BUILD OS      = darwin

  CC            = /usr/local/bin/ccache gcc
  CFLAGS        = -g -O2
  CPPFLAGS      = -Qunused-arguments  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -I/usr/local/opt/berkeley-db4/include -DMAC_OSX
  CXX           = /usr/local/bin/ccache g++ -std=c++11
  CXXFLAGS      = -g -O2 -Wall -Wextra -Wformat -Wformat-security -Wno-unused-parameter -Wno-self-assign -Wno-unused-local-typedef -Wno-deprecated-register
  LDFLAGS       =  -Wl,-headerpad_max_install_names -Wl,-dead_strip
```

jonasschnelli committed:
Add compile and link options echo to configure

luke-jr commented:
What's with all the uppercase? Probably should specify which version of Qt.

laanwj commented:
Concept ACK

MarcoFalke commented:
utACK 9d380ea

laanwj commented:
@cfields can you take a look as well?

laanwj commented:
@theuni can you take a look as well?

paveljanik commented on configure.ac:
```diff
@@ -1126,3 +1126,22 @@ case ${OS} in
    ;;
 esac
+
+echo ""
```
No need for `""`.

paveljanik commented on configure.ac:
```diff
@@ -1126,3 +1126,22 @@ case ${OS} in
+echo "Options used to compile and link:"
+echo "  WITH WALLET   = $enable_wallet"
+echo "  WITH GUI / QT = $bitcoin_enable_qt"
+echo "   WITH QR      = $use_qr"
```
"Conditionalize" QR on QT?

paveljanik commented:
Can you add ZMQ and miniupnpc co make it complete?

paveljanik commented:
Concept ACK!

jonasschnelli commented:
Updated, added major Qt version, ZMQ, UPNP, DEBUG ENABLED (more obvious then checking for -O0).
QR CODE and QT VERSION is only visible when QT is enabled.

This is how it looks now:
```
Options used to compile and link:
  WITH WALLET   = yes
  WITH GUI / QT = yes
    QT VERSION  = 5
    WITH QR     = yes
  WITH ZMQ      = yes
  WITH TEST     = yes
  WITH BENCH    = yes
  WITH UPNP     = yes
  DEBUG ENABLED = no

  TARGET OS     = darwin
  BUILD OS      = darwin

  CC            = /usr/local/bin/ccache gcc
  CFLAGS        = -g -O2
  CPPFLAGS      = -Qunused-arguments  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -I/usr/local/opt/berkeley-db4/include -DMAC_OSX
  CXX           = /usr/local/bin/ccache g++ -std=c++11
  CXXFLAGS      = -g -O2 -Wall -Wextra -Wformat -Wformat-security -Wno-unused-parameter -Wno-self-assign -Wno-unused-local-typedef -Wno-deprecated-register
  LDFLAGS       =  -Wl,-headerpad_max_install_names -Wl,-dead_strip

```

sipa commented:
utACK ce1acfc6e9fb6e81c11f3d65446999462206eae6

paveljanik commented:
There are too many uppercase words WITH. I think that only variable names (`CC`, `CFLAGS`, ..., `LDFLAGS`) should be uppercased.

theuni commented:
Very nice, concept ACK. Will review.

laanwj commented:
> There are too many uppercase words WITH. I think that only variable names (CC, CFLAGS, ..., LDFLAGS) should be uppercased.

I tend to agree. Not that I care much for aesthetic reasons, but it can result in confusion, in the `./configure --help` interface uppercase means 'a variable that can be overridden'. So yes let's make it less yelly :-)

jonasschnelli commented:
Switched the non CFLAGish stuff to lowercase:

This is how it looks now:
```
Options used to compile and link:
  with wallet   = yes
  with gui / qt = yes
    qt version  = 5
    with qr     = yes
  with zmq      = yes
  with test     = yes
  with bench    = yes
  with upnp     = yes
  debug enabled = no

  target os     = darwin
  build os      = darwin

  CC            = /usr/local/bin/ccache gcc
  CFLAGS        = -g -O2
  CPPFLAGS      = -Qunused-arguments  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -I/usr/local/opt/berkeley-db4/include -DMAC_OSX
  CXX           = /usr/local/bin/ccache g++ -std=c++11
  CXXFLAGS      = -g -O2 -Wall -Wextra -Wformat -Wformat-security -Wno-unused-parameter -Wno-self-assign -Wno-unused-local-typedef -Wno-deprecated-register
  LDFLAGS       =  -Wl,-headerpad_max_install_names -Wl,-dead_strip
```

fanquake commented:
Travis errors are unrelated. Output looks ok.

```
config.status: executing libtool commands

Options used to compile and link:
  with wallet   = yes
  with gui / qt = yes
    qt version  = 5
    with qr     = yes
  with zmq      = yes
  with test     = yes
  with bench    = yes
  with upnp     = yes
  debug enabled = no

  target os     = darwin
  build os      = darwin

  CC            = gcc
  CFLAGS        = -g -O2
  CPPFLAGS      =  -DHAVE_BUILD_INFO -D__STDC_FORMAT_MACROS -I/usr/local/opt/berkeley-db4/include -DMAC_OSX
  CXX           = g++ -std=c++11
  CXXFLAGS      = -g -O2 -Wall -Wextra -Wformat -Wformat-security -Wno-unused-parameter -Wno-self-assign -Wno-unused-local-typedef -Wno-deprecated-register
  LDFLAGS       =  -Wl,-headerpad_max_install_names -Wl,-dead_strip
```

paveljanik commented on configure.ac:
```diff
@@ -1126,3 +1126,28 @@ case ${OS} in
+echo "  with wallet   = $enable_wallet"
+echo "  with gui / qt = $bitcoin_enable_qt"
+if test x$bitcoin_enable_qt != xno; then
+echo "    qt version  = $bitcoin_qt_got_major_vers"
```
indent please.

paveljanik commented on configure.ac:
```diff
@@ -1126,3 +1126,28 @@ case ${OS} in
+echo 
+echo "  target os     = $TARGET_OS"
+echo "  build os      = $BUILD_OS"
+echo ""
```
no need for ""

paveljanik commented:
ACK https://github.com/bitcoin/bitcoin/pull/9156/commits/d8274bcd00865d829808e8e05651c61dff7ccac0
